### PR TITLE
Deobfuscate stacktraces in log messages using a RewriteAppender and a custom RewritePolicy

### PR DESCRIPTION
**Pull request by jpenilla**

### CODE REVIEW

File: patches/server/0026-Prevent-tile-entity-and-entity-crashes.patch [hunk 1]
```diff
@@ -18,8 +18,8 @@ index d29334b5dee6fe3f292f774442304b7dc6b0e2f0..d6ed061e36668ea1cbaee44c06575534
 -            entity.fillCrashReportCategory(crashreportsystemdetails);
 -            throw new ReportedException(crashreport);
 +            // Paper start - Prevent tile entity and entity crashes
-+            System.err.println("Entity threw exception at " + entity.level.getWorld().getName() + ":" + entity.getX() + "," + entity.getY() + "," + entity.getZ());
-+            throwable.printStackTrace();
++            final String msg = String.format("Entity threw exception at %s:%s,%s,%s", entity.level.getWorld().getName(), entity.getX(), entity.getY(), entity.getZ());
++            MinecraftServer.LOGGER.error(msg, throwable);
```
Comment (Proximyst) [inline patches/server/0026-Prevent-tile-entity-and-entity-crashes.patch:22]: msg could be inlined and therefore use the log4j `{}` syntax

File: patches/server/0717-Deobfuscate-stacktraces-in-log-messages-using-a-Rewr.patch [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+diff --git a/src/main/resources/log4j2.xml b/src/main/resources/log4j2.xml
+index 67da1aa7a21622fb231d19dede3775a282a4a12e..f91d0df569b2f1d430ea5eee5f53779902a4f32c 100644
+--- a/src/main/resources/log4j2.xml
++++ b/src/main/resources/log4j2.xml
+@@ -29,15 +29,19 @@
+             </Policies>
+             <DefaultRolloverStrategy max="1000"/>
+         </RollingRandomAccessFile>
++        <Rewrite name="rewrite">
++            <StacktraceDeobfuscatingRewritePolicy />
```
Comment (Proximyst) [inline patches/server/0717-Deobfuscate-stacktraces-in-log-messages-using-a-Rewr.patch:367]: Can we make it possible to opt-out somehow?

Comment (jpenilla) [reply patches/server/0717-Deobfuscate-stacktraces-in-log-messages-using-a-Rewr.patch:367]: Can use a custom log4j config, but, could also add an option in paper config to make it do nothing if that's what you mean.

File: patches/server/0026-Prevent-tile-entity-and-entity-crashes.patch (hunk 1 above)
Comment (jpenilla) [reply patches/server/0026-Prevent-tile-entity-and-entity-crashes.patch:22]: Yeah, had it doing that, however a later patch uses the string given to the logger for the exception event

File: patches/server/0717-Deobfuscate-stacktraces-in-log-messages-using-a-Rewr.patch (hunk 2 above)
Comment (MiniDigger) [reply patches/server/0717-Deobfuscate-stacktraces-in-log-messages-using-a-Rewr.patch:367]: Was thinking about that too. NMS devs will be super confused, so at least give them a easy opt out somehow, custom log4j config is a bit meh

**Comment by jpenilla** [comment]
Added a config option to disable stacktrace deobfuscation without needing to use a custom log4j config

**Comment by DenWav** [comment]
This isn't how I would impermanent the Gradle config, but it's too late for me to open my computer and type something up for a replacement, so I'll need to so that tomorrow.

**Review by Proximyst — APPROVED** [review]
Fine to me. Wait for Den's suggestions on Gradle conf

**Comment by DenWav** [comment]
After reviewing I see that thus Gradle config is basically all good as we'll get due to how paperweight sets these tasks up. I'll update paperweight to be better here, but for now let's just merge this.

**Pull request merged**